syft/format/internal/spdxutil/helpers/download_location.go

```diff
@@ -14,21 +17,38 @@ func DownloadLocation(p pkg.Package) string {
 	//   (ii) the SPDX file creator has made no attempt to determine this field; or
 	//   (iii) the SPDX file creator has intentionally provided no information (no meaning should be implied by doing so).
 
+	var location string
 	if hasMetadata(p) {
 		switch metadata := p.Metadata.(type) {
 		case pkg.ApkDBEntry:
-			return NoneIfEmpty(metadata.URL)
+			location = metadata.URL
 		case pkg.NpmPackage:
-			return NoneIfEmpty(metadata.URL)
+			location = metadata.URL
 		case pkg.NpmPackageLockEntry:
-			return NoneIfEmpty(metadata.Resolved)
+			location = metadata.Resolved
 		case pkg.PhpComposerLockEntry:
-			return NoneIfEmpty(metadata.Dist.URL)
+			location = metadata.Dist.URL
 		case pkg.PhpComposerInstalledEntry:
-			return NoneIfEmpty(metadata.Dist.URL)
+			location = metadata.Dist.URL
 		case pkg.OpamPackage:
-			return NoneIfEmpty(metadata.URL)
+			location = metadata.URL
 		}
 	}
-	return NOASSERTION
+	return UriValue(location)
+}
+
+func isUriValid(uri string) bool {
+	_, err := urilib.NewURIRef(uri)
+	return err == nil
+}
+
+func UriValue(uri string) string {
+	if NoneIfEmpty(uri) != NONE {
```

Comment:
I think using `NOASSERTION` here actually ends up being the right thing to do. The reason is, we're not claiming that there is no download location, we're only claiming that we cannot provide one. As such, I think we should update this to: `if isURIValid(uri) { return uri } return NOASSERTION`, or instead get rid of this function altogether and just update the last bit of `DownloadLocation` to have this, and ensure `isURIValid` returns false for empty string. I'm happy to make these updates.

Reply:
So if I understand it correctly, NONE should not be included here if it's an empty string. It is however a valid option according to the spec. Should there be a clause to set the download location to NONE if it is set as such by the creator of the package?

From the spec:

> NONE if there is no download location whatsoever.

I'll make the changes to set to NOASSERTION, unless a package explicitly sets the download location to NONE. Let me know what you think :)

Reply:
Added some changes suggested by `make lint-fix` as well.